trace/sql/sql.go

```diff
@@ -373,3 +386,33 @@ func (tx *Tx) Stmt(ctx context.Context, stmt *Stmt) *Stmt {
 	defer trace.SpanSuccess(sp)
 	return &Stmt{stmt: tx.tx.StmtContext(ctx, stmt.stmt)}
 }
+
+func parseDSN(dsn string) DSNInfo {
+	res := DSNInfo{}
+
+	dsnPattern := regexp.MustCompile(
```

Comment:
would it make sense to avoid parsing the regexp on each call ?
e.g. refactor it out of the func, or embed in some helper struct ?